controller/test-files/work_item/list_children/ok.res.payload.golden.json

```diff
@@ -77,7 +88,7 @@
             "self": "http:///api/spaces/00000000-0000-0000-0000-000000000004"
           }
         },
-        "system.boardcolumns": {},
+        "system_boardcolumns": {},
```

Comment:
Why is this replaced when all other fields are just added?

Reply:
Thank you so much. This was changed because we the `addNewFieldNames` method adds fields only to the attributes. `boardcolumns`, `area`, `iteration`, `assignee` and `labels` are all part of the `relationships`. I'll fix this. :+1:

Reply:
We can't add `system_*` fields to relationships. The `workitemRelationships` struct is defined as
```go
// WorkItemRelationships defines only `assignee` as of now. To be updated
var workItemRelationships = a.Type("WorkItemRelationships", func() {
	a.Attribute("assignees", relationGenericList, "This defines assignees of the Work Item")
	a.Attribute("labels", relationGenericList, "List of labels attached to the Work Item")
	a.Attribute("system_boardcolumns", relationGenericList, "List of board columns this Work Item is attached to")
	a.Attribute("creator", relationGeneric, "This defines creator of the Work Item")
	a.Attribute("baseType", relationBaseType, "This defines type of Work Item")
	a.Attribute("comments", relationGeneric, "This defines comments on the Work Item")
	a.Attribute("iteration", relationGeneric, "This defines the iteration this work item belong to")
	a.Attribute("area", relationGeneric, "This defines the area this work item belongs to")
	a.Attribute("children", relationGeneric, "This defines the children of this work item")
	a.Attribute("space", relationSpaces, "This defines the owning space of this work item.")
	a.Attribute("parent", relationKindUUID, "This defines the parent of this work item.")
	a.Attribute("workItemLinks", relationGeneric, "List of links in which this work item is involved")
	a.Attribute("events", relationGeneric, "List of events in which this work item is involved")
})
```

I ran my code against UI and it looks like nothing breaks if the relationships are directly renamed to `system_*`. The update on these fields also works fine.
@sudsen/@sahil143 can you guys confirm that you use relationship field name from the workitem types? That is if the relationship has `system_iteration` instead of `system.iteration` nothing would break on the UI.

Reply:
It'll break things in UI. We have to make changes accordingly in UI.

Reply:
Based on https://chat.openshift.io/developers/pl/e9fe383wjtynjymu557nqg9ija
>Okay, that won't break anything

Reply:
@kwk Shall we rename `system.boardcolumns` to `boardcolumns` in the relationships? Then we won't have this problem.

Reply:
> @kwk Shall we rename `system.boardcolumns` to `boardcolumns` in the relationships? Then we won't have this problem.

@jarifibrahim but what about the current board implementation. It must read boardcolumns from `system.boardcolumns`, no?

Reply:
> > @kwk Shall we rename `system.boardcolumns` to `boardcolumns` in the relationships? Then we won't have this problem.
> 
> @jarifibrahim but what about the current board implementation. It must read boardcolumns from `system.boardcolumns`, no?

@kwk No. See this block of code  
https://github.com/fabric8-services/fabric8-wit/blob/9dc1c08d14ba582ae68ba5e27551ab6262258043/controller/workitem.go#L376-L402

We convert the `system.boardcolumn` relationship to `workitem.attibute[systemboardcolumns]`. I'll just update https://github.com/fabric8-services/fabric8-wit/blob/master/design/workitems.go#L38 line and everything should work as expected. We might have a lot of golden file changes :)

Reply:
I trust you with this and if the board view in planner goes nuts I don't care that much as it is probably not finished anyways.

Reply:
I've raised https://github.com/fabric8-services/fabric8-wit/pull/2345 for it.